### PR TITLE
[VC-43403] CyberArk(agent): add support for MachineHub output mode

### PR DESCRIPTION
**Pull request by wallrj-cyberark**
CyberArk(agent): add support for MachineHub output mode:

- Introduced a new `MachineHub` output mode in the agent configuration.
- Added `--machine-hub` flag to enable the `MachineHub` mode.
- Implemented `CyberArkClient` for publishing data readings to CyberArk's API.
- Created `LoadClientConfigFromEnvironment` to load CyberArk client configuration from environment variables.
- Updated tests to cover `MachineHub` mode and CyberArk client functionality.
- Modified mock data and discovery logic to support CyberArk integration.

## Followup PRs
* #684
* #687 
* #678 

## Testing
This gets us to the point of being able to run the agent in machine hub mode from the command line, as follows:

```bash
export ARK_USERNAME=*****
export ARK_SECRET=*****
export ARK_SUBDOMAIN=tlskp-test
export ARK_DISCOVERY_API=https://platform-discovery.integration-cyberark.cloud/api/v2
```

```bash
go run . agent --one-shot --machine-hub -v 6 --agent-config-file ./examples/machinehub.yaml
```

```
I0828 17:58:04.691982  797055 run.go:58] "Starting" logger="Run" version="development" commit=""
I0828 17:58:04.692183  797055 config.go:448] "Output mode selected" logger="Run" mode="MachineHub" reason="--machine-hub was specified"
I0828 17:58:04.692208  797055 run.go:116] "Healthz endpoints enabled" logger="Run.APIServer" addr=":8081" path="/healthz"
I0828 17:58:04.692226  797055 run.go:120] "Readyz endpoints enabled" logger="Run.APIServer" addr=":8081" path="/readyz"
I0828 17:58:04.692246  797055 run.go:269] "Pod event recorder disabled" logger="Run" reason="The agent does not appear to be running in a Kubernetes cluster." detail="When running in a Kubernetes cluster the following environment variables must be set: POD_NAME, POD_NODE, POD_UID, POD_NAMESPACE"
I0828 17:58:04.692256  797055 run.go:185] "Starting DataGatherer" logger="Run" name="dummy"
I0828 17:58:04.692266  797055 run.go:374] "Successfully gathered" logger="Run.gatherAndOutputData.gatherData" name="dummy"
I0828 17:58:04.757339  797055 run.go:432] "Starting" logger="Run.APIServer.ListenAndServe" addr=":8081"
I0828 17:58:04.973449  797055 round_trippers.go:632] "Response" logger="Run.gatherAndOutputData.postData" verb="GET" url="https://platform-discovery.integration-cyberark.cloud/api/v2/services/subdomain/tlskp-test" status="200 OK" milliseconds=281
I0828 17:58:05.308928  797055 round_trippers.go:632] "Response" logger="Run.gatherAndOutputData.postData" verb="POST" url="https://anb5751.id.integration-cyberark.cloud/Security/StartAuthentication" status="200 OK" milliseconds=333
I0828 17:58:05.309544  797055 identity.go:303] "made successful request to StartAuthentication" logger="Run.gatherAndOutputData.postData" source="Identity.doStartAuthentication" summary="NewPackage"
I0828 17:58:05.615312  797055 round_trippers.go:632] "Response" logger="Run.gatherAndOutputData.postData" verb="POST" url="https://anb5751.id.integration-cyberark.cloud/Security/AdvanceAuthentication" status="200 OK" milliseconds=304
I0828 17:58:05.615731  797055 identity.go:419] "successfully completed AdvanceAuthentication request to CyberArk Identity; login complete" logger="Run.gatherAndOutputData.postData" username="..."
I0828 17:58:06.158647  797055 round_trippers.go:632] "Response" logger="Run.gatherAndOutputData.postData" verb="POST" url="https://tlskp-test.inventory.integration-cyberark.cloud/api/ingestions/kubernetes/snapshot-links" status="200 OK" milliseconds=542
I0828 17:58:06.600752  797055 round_trippers.go:632] "Response" logger="Run.gatherAndOutputData.postData" verb="PUT" url="..." status="200 OK" milliseconds=441
I0828 17:58:06.600885  797055 run.go:417] "Data sent successfully" logger="Run.gatherAndOutputData.postData"
I0828 17:58:06.600918  797055 run.go:446] "Shutting down" logger="Run.APIServer.ListenAndServe" addr=":8081"
I0828 17:58:06.601070  797055 run.go:461] "Shutdown complete" logger="Run.APIServer.ListenAndServe" addr=":8081"
```

### CODE REVIEW

File: pkg/agent/config.go [hunk 1]
```diff
@@ -334,6 +334,7 @@ const (
 	VenafiCloudKeypair          OutputMode = "Venafi Cloud Key Pair Service Account"
 	VenafiCloudVenafiConnection OutputMode = "Venafi Cloud VenafiConnection"
 	LocalFile                   OutputMode = "Local File"
+	MachineHub                  OutputMode = "MachineHub"
```
Comment (inteon) [inline pkg/agent/config.go:340]: I think we need a way to select multiple output modes.

Comment (inteon) [reply pkg/agent/config.go:340]: So we can combine (something Venafi, Local File and MachineHub).

Comment (maelvls) [reply pkg/agent/config.go:340]: From Richard:

> Atanas has agreed that the first version of this new agent does not need to upload to both TLSPK and MachineHub, so that simplifies things quite a bit.

This enum is enough for now.

File: pkg/client/client_cyberark_test.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	_ "k8s.io/klog/v2/ktesting/init"
+)
+
+// TestCyberArkClient_PostDataReadingsWithOptions_RealAPI demonstrates that the
+// dataupload code works with the real inventory API.
+//
+// To enable verbose request logging:
+//
+//	go test ./pkg/internal/cyberark/dataupload/... \
+//	  -v -count 1 -run TestPostDataReadingsWithOptionsWithRealAPI -args -testing.v 6
```
Comment (maelvls) [inline pkg/client/client_cyberark_test.go]: How do I test this myself? What values should I use for `ARK_PLATFORM_DOMAIN` and the others?

Comment (wallrj-cyberark) [reply pkg/client/client_cyberark_test.go]: I've shared the test tenant and credentials in a DM.

File: pkg/internal/cyberark/dataupload/dataupload.go [hunk 3]
```diff
@@ -35,10 +35,6 @@ type CyberArkClient struct {
 	authenticateRequest func(req *http.Request) error
 }
 
-type Options struct {
-	ClusterName string
-}
-
```
Comment (wallrj-cyberark) [inline pkg/internal/cyberark/dataupload/dataupload.go:41]: Forgot to remove this in #703

File: pkg/internal/cyberark/dataupload/dataupload_test.go [hunk 4]
```diff
@@ -109,47 +102,3 @@ func TestCyberArkClient_PutSnapshot_MockAPI(t *testing.T) {
...
-	identityClient := identity.New(httpClient, services.Identity.API, subdomain)
-	err = identityClient.LoginUsernamePassword(ctx, username, []byte(secret))
-	require.NoError(t, err)
-
-	cyberArkClient := dataupload.New(httpClient, services.DiscoveryContext.API, identityClient.AuthenticateRequest)
-	err = cyberArkClient.PutSnapshot(ctx, dataupload.Snapshot{
-		ClusterID: "bb068932-c80d-460d-88df-34bc7f3f3297",
-	})
-	require.NoError(t, err)
-}
```
Comment (wallrj-cyberark) [inline pkg/internal/cyberark/dataupload/dataupload_test.go:155]: The code used to setup the dataupload wrapper in this test has now been moved to `pkg/internal/cyberark`, so I've moved this test there and updated it to simply use the new `NewDatauploadClient` function.

File: pkg/client/client_cyberark_test.go
Comment (wallrj-cyberark) [inline pkg/client/client_cyberark_test.go:1]: I'll add more unit tests here in #684 when I implement the conversion between DataReadings and Snapshot.
In a future PR I might also remove the "RealAPI" test which mostly duplicates the same test in `pkg/internal/cyberark/client_test.go`.

File: pkg/internal/cyberark/identity/mock.go [hunk 5]
```diff
@@ -28,7 +28,7 @@ const (
 	// mockSuccessfulStartAuthenticationToken is the token returned by the
 	// mock server in response to a successful AdvanceAuthentication request
 	// Must match what's in testdata/advance_authentication_success.json
-	mockSuccessfulStartAuthenticationToken = "long-token"
+	mockSuccessfulStartAuthenticationToken = "success-token"
```
Comment (wallrj-cyberark) [inline pkg/internal/cyberark/identity/mock.go:31]: All the fake values in the various mocks have to be consistent for the `testutil.FakeCyberArk` combined mock to work.
In a future PR I might try and fix that.

File: pkg/testutil/envtest.go [hunk 6]
```diff
@@ -259,6 +262,33 @@ func FakeTPP(t testing.TB) (*httptest.Server, *x509.Certificate) {
...
 }
 
+// FakeCyberArk returns an HTTP client that will route requests to mock CyberArk
+// Service Discovery, Identity and Discovery and Context APIs. This is useful
+// for testing code that uses all those APIs, such as
+// `cyberark.NewDatauploadClient`.
+//
+// The environment variable `ARK_DISCOVERY_API` is set to the URL of the mock
+// Service Discovery API, for the supplied `testing.TB` so that the client under
+// test will use the mock Service Discovery API.
```
Comment (maelvls) [inline pkg/testutil/envtest.go:272]: Not a big deal, but `ARK_DISCOVERY_API` seems to default to `https://platform-discovery.integration-cyberark.cloud/api/v2`, but doesn't default to anything in the mock... I found that because I had forgotten to set `ARK_DISCOVERY_API`, and was super confused by the following output:

Without `ARK_DISCOVERY_API`:

```
I0829 10:39:18.377748   32025 round_trippers.go:632] "Response" logger="Run.gatherAndOutputData.postData" verb="GET" url="https://platform-discovery.cyberark.cloud/api/v2/services/subdomain/tlskp-test" status="404 Not Found" milliseconds=1063
I0829 10:39:18.377812   32025 run.go:334] "Warning: PushingErr: retrying" logger="Run.gatherAndOutputData" in="22.148195504s" reason="post to server failed: while initializing data upload client: got an HTTP 404 response from service discovery; maybe the subdomain \"tlskp-test\" is incorrect or does not exist?"
```

With `ARK_DISCOVERY_API`:

```
I0829 10:40:45.658647   33943 round_trippers.go:632] "Response" logger="Run.gatherAndOutputData.postData" verb="GET" url="https://platform-discovery.integration-cyberark.cloud/api/v2/services/subdomain/tlskp-test" status="200 OK" milliseconds=212
I0829 10:40:46.155025   33943 round_trippers.go:632] "Response" logger="Run.gatherAndOutputData.postData" verb="POST" url="https://anb5751.id.integration-cyberark.cloud/Security/StartAuthentication" status="200 OK" milliseconds=495
```

IMO the mock should either use a default `ARK_DISCOVERY_API` value if missing, or it should fail and tell the user to set `ARK_DISCOVERY_API`. I might have misunderstood though.
